### DISCUSSION TITLE
Underrepresentation of Dev Tools

### DISCUSSION THREAD

**Discussion by KatsuNoxin** [cardano-foundation/developer-portal#891] [Ideas]
Hi, I've been forwarded here to drop an idea. 

**Problem:** 
Dev tools are quite daunting for non-devs to understand, and most don't really understand what they do. This leads to less exposure to the wider community. The community understands Hydra, Input Endorsers, Pipelining, etc to a degree and knows they're there for scaling and that gets everyone to share these on social media. But, I've noticed the lack of sharing tools and languages like let's say Aiken and Kupo to the wider community. So with less exposure to such tools from the community, there will be less exposure to developers as well. This doesn't mean developers won't find them, but it's not as easy as it could be. I think we should want to ease this process as Cardano scales. It could also lead to new contributors to these tools if they find it interesting, but they have to find it first.

**Why:**
I believe it's because of the lack of education on Cardano's tools and etc. How many of the community would understand what cardano-db-sync does and why tools like Kupo, Oura, and Carp matter in the same vein? 

**What can be done:**
Make a sister site for developers.cardano.org or make it more non-dev friendly. 

**How:**
Create content that allows the non-dev community to understand the tools and inner workings so that they can participate in sharing them. 

> "The purpose of Cardano DB Sync is to follow the Cardano chain and take information from the chain and an internally maintained copy of ledger state. Data is then extracted from the chain and inserted into a PostgreSQL database. SQL queries can then be written directly against the database schema or as queries embedded in any language with libraries for interacting with an SQL database."

Most non-devs would somewhat understand the first half, but get lost in the second. 

> "Carp is a new indexer for Cardano that saves its data to a Postgres database (SQL). If you’re not a programmer and don’t know what this means, we’ll briefly introduce this to you." 

This is an excerpt I pulled from Sebastian's medium post about Carp, which he follows up with an explanation of what an indexer is. 

> "Blockchains contain a lot of information — often hundreds of gigabytes worth of data. Applications like dApps and wallets need to quickly filter through this data to provide the fast user experience that users expect. This is what indexers are meant to solve. There are many different kinds of indexers and the best one for the job tends to depend on what you’re trying to do. SQL databases are a popular kind of indexer because they provide a lot of flexibility to users to query what they want while maintaining decent performance."

I believe there's a need for more content like this that's easily digested and can be shared by the community. Especially on social media like Twitter and Reddit. Begin with the core tools like cardano-db-sync then on to newly created tools by the community. Starting from the bottom to top. 

As we head towards being a self-governed blockchain, I think education like this becomes more important. I think the whole Daedalus Turbo debacle would be just one example. If the community is left out, there's the issue of whether they have the knowledge needed to self-govern the development side of Cardano. dRep's might be a solution, but not something we can wholly rely on.

I would love to volunteer as a writer for this idea.

**Comment by rdlrt** [comment]
While simplifying where possible is good, the suggestion here is not really clear (being OK with folks who are going to work on querying data from a database not knowing SQL at all is a bit iffy - as that'd likely result in pretty bad quality/performance projects, which are in abundance already in crypto ecosystems, but that's just my personal opinion). Also, I am not sure if having yet-another-site is helpful, the idea is to consolidate - not distribute further.

Instead, you can perhaps post suggestions/edits to the existing pages to make it more easier/friendly to read. You're always welcome to submit content updates on any of the pages following the [contribution guidelines](https://developers.cardano.org/docs/portal-contribute/).

**Reply by KatsuNoxin** [reply]
I think you didn't get my point. Dev tools are lacking in exposure, this could, and maybe already did lead developers outside of Cardano to believe one can only build using Plutus and Haskell. The idea is to educate non-devs on basic concepts to allow them to understand what dev tools are for. This could lead to more exposure as we head to self-governing this is important as well because voters need to understand what they're voting for.

**Comment by OlofBlomqvist** [comment]
> Dev tools are lacking in exposure, this could, and maybe already did lead developers outside of Cardano to believe one can only build using Plutus and Haskell.

Aiken, eopsin among others will very soon have their own pages under the smart contract category which should help to make it clear that Cardano is not limited to Plutus and Haskell.

> Make a sister site for developers.cardano.org or make it more non-dev friendly.

Perhaps some of the content you want to make available might fit better over http://docs.cardano.com (https://github.com/input-output-hk/cardano-documentation)?

**Reply by rphair** [reply]
ref:

- https://github.com/cardano-foundation/developer-portal/pull/926
- https://github.com/cardano-foundation/developer-portal/pull/928
- https://github.com/cardano-foundation/developer-portal/pull/929
